test/Microsoft.ComponentDetection.Orchestrator.Tests/Services/DefaultGraphTranslationServiceTests.cs

```diff
@@ -0,0 +1,100 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+[TestClass]
+[TestCategory("Governance/All")]
+[TestCategory("Governance/ComponentDetection")]
+public class DefaultGraphTranslationServiceTests
+{
+    private readonly DefaultGraphTranslationService serviceUnderTest;
+    private readonly ContainerDetails sampleContainerDetails;
+    private readonly ComponentRecorder componentRecorder;
+    private readonly Mock<IComponentDetector> componentDetectorMock;
+    private readonly DirectoryInfo sourceDirectory;
+
+    public DefaultGraphTranslationServiceTests()
+    {
+        this.serviceUnderTest = new DefaultGraphTranslationService(new Mock<ILogger<DefaultGraphTranslationService>>().Object);
+        this.componentRecorder = new ComponentRecorder(new Mock<ILogger>().Object);
+
+        this.sampleContainerDetails = new ContainerDetails { Id = 1 };
+        this.componentDetectorMock = new Mock<IComponentDetector>();
+        this.componentDetectorMock.SetupGet(x => x.Id).Returns("Detector1");
+        this.componentDetectorMock.SetupGet(x => x.Version).Returns(1);
+        this.sourceDirectory = new DirectoryInfo(Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()));
+        this.sourceDirectory.Create();
+    }
+
+    [TestMethod]
+    public void GenerateScanResultFromResult_WithCustomLocations()
+    {
+        var detectedFilePath = "/some/file/path";
+        var npmCustomPath = "/custom/path.js";
+        var nugetCustomPath = "/custom/path2.csproj";
+        var relatedFilePath = "/generic/relevant/path";
+
+        var singleFileComponentRecorder = this.componentRecorder.CreateSingleFileComponentRecorder(Path.Join(this.sourceDirectory.FullName, detectedFilePath));
+        var processingResult = new DetectorProcessingResult
+        {
+            ResultCode = ProcessingResultCode.Success,
+            ContainersDetailsMap = new Dictionary<int, ContainerDetails>
+            {
+                {
+                    this.sampleContainerDetails.Id, this.sampleContainerDetails
+                },
+            },
+            ComponentRecorders = new[] { (this.componentDetectorMock.Object, this.componentRecorder) },
+        };
+
+        var expectedNpmComponent = new NpmComponent("npm-component", "1.2.3");
+        var expectedNugetComponent = new NuGetComponent("nugetComponent", "4.5.6");
+        var detectedNpmComponent = new DetectedComponent(expectedNpmComponent);
+        var detectedNugetComponent = new DetectedComponent(expectedNugetComponent);
+
+        // Any Related File will be reported for ALL components found in this graph
+        singleFileComponentRecorder.AddAdditionalRelatedFile(Path.Join(this.sourceDirectory.FullName, relatedFilePath));
+
+        // Registering components in same manifest with different custom paths
+        detectedNpmComponent.AddComponentFilePath(Path.Join(this.sourceDirectory.FullName, npmCustomPath));
```

Comment:
No detector in this repo calls this `AddComponentFilePath` directly, as the file location gets automatically added by the graph. It should be a NoOp change for this repo.